File: app/src/main/java/edu/uark/cartapp/models/api/fields/ProductFieldName.java
package edu.uark.cartapp.models.api.fields;

import edu.uark.cartapp.models.api.interfaces.FieldNameInterface;

/* ==== APP ProductFieldName.java ====*/
public enum ProductFieldName implements FieldNameInterface {
	ID("id"),
	LOOKUP_CODE("lookupCode"),
	COUNT("count"),
	API_REQUEST_STATUS("apiRequestStatus"),
	API_REQUEST_MESSAGE("apiRequestMessage"),
	CREATED_ON("createdOn");

	private String fieldName;

	ProductFieldName(String fieldName) {
		this.fieldName = fieldName;
	}

	public String getFieldName() { return this.fieldName; }
}

File: app/src/main/java/edu/uark/cartapp/models/api/fields/ProductFieldName.kt
package edu.uark.cartapp.models.api.fields

import edu.uark.cartapp.models.api.interfaces.FieldNameInterface

/* ==== APP ProductFieldName.java ====*/
enum class ProductFieldName private constructor(private val fieldName: String) : FieldNameInterface {
	ID("id"),
	LOOKUP_CODE("lookupCode"),
	COUNT("count"),
	API_REQUEST_STATUS("apiRequestStatus"),
	API_REQUEST_MESSAGE("apiRequestMessage"),
	CREATED_ON("createdOn");

	override fun getFieldName(): String {
		return this.fieldName
	}
}